components/brave_wallet_ui/page/screens/fund-wallet/deposit-funds.tsx

```diff
@@ -550,6 +558,18 @@ function DepositAccount() {
     [copyAddressToClipboard]
   )
 
+  React.useEffect(() => {
+    ;(async () => {
+      if (!selectedAccount) {
+        return
+      }
+      const address = await generateReceiveAddress(
+        selectedAccount.accountId
+      ).unwrap()
+      setReceiveAddress(address)
+    })()
+  }, [selectedAccount])
```

Comment:
Potential memory leak. Must check that the component is mounted before setting state.

```suggestion
  React.useEffect(() => {
    let ignore = false
    ;(async () => {
      if (!selectedAccount) {
        return
      }
      const address = await generateReceiveAddress(
        selectedAccount.accountId
      ).unwrap()
      if (!ignore) {
        setReceiveAddress(address)
      }
    })()
    
    // cleanup
    return () => {
      ignore = true
    }
  }, [selectedAccount])
```